### ISSUE TITLE
Arrow keys and navigation keys do not work inside a container when using ConEmu as a console on Windows

### ISSUE THREAD

**Issue by MatthewMiele** [docker/docker#25475]
**Output of `docker version`:**

```
Docker version 1.12.0, build 8eab29e
```

**Output of `docker info`:**

```
Containers: 32
 Running: 2
 Paused: 0
 Stopped: 30
Images: 28
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 108
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host null overlay bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.952 GiB
Name: moby
ID: SUPY:2DKV:N3C2:ZPXD:AZQ5:DQIQ:5I7W:XFEN:M2YV:Y2VY:L3EO:FAZC
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
Windows 10 with Anniversary update
ConEmu 160724 [64]
```

**Steps to reproduce the issue:**
1. Open ConEmu using regular windows CMD as shell.
2. Start a docker container and then enter into it. Such as docker exec -it CONTAINER /bin/bash
3. Type a command and hit enter. i.e ```ls``
4. Press the up arrow on the keyboard
5. Enter a command (without hitting enter at the end)
6. Press the home key
7. Press the left and right arrows

**Describe the results you received:**
After step 4, the line remains blank, the previous command is not shown
After step 6, the cursor remains at the end of the line, not moving to the start
After step 7: the cursor remains at the end of the line, no movement

**Describe the results you expected:**
After step 4, expected to see the last command typed. i.e ```ls``` from step 3
After step 6, expected the cursor to be at the start of the line
after step 7, expected the cursor to move left and right through the command

**Additional information you deem important (e.g. issue happens only occasionally):**
When first researching this issue I came across a previously raised issue on the exact problem which is now closed, as a fix has been made and merged in.
Previous issue: https://github.com/docker/docker/issues/13817
Previous fix: https://github.com/docker/docker/pull/19823

@mdunc I have raised this as a new issue
@Maximus5 found and made the original fix

**Comment by thaJeztah** [comment]
ping @jstarks ptal

**Comment by MatthewMiele** [comment]
As mentioned in my other bug report, @rwilliams found that commit 4acc2c7 seems to be the commit that's added this regression

**Comment by rwilliams** [comment]
Forcing ANSI emulation on the input stream seems to fix the issue but I'm not if it's the right thing to always have the input stream emulated. see https://github.com/rwilliams/docker/commit/7313ed9316a91c03de3b39aa4cc5860ae6cb7515

**Comment by jstarks** [comment]
Probably this is a reasonable workaround for now, until conemu and Windows start playing nicely together again. I will take a closer look in the morning.

**Comment by jstarks** [comment]
Cc @jhowardmsft
